src/workerpool.jl

```diff
@@ -135,6 +135,28 @@ function remotecall_pool(rc_f, f, pool::AbstractWorkerPool, args...; kwargs...)
     end
 end
 
+# Specialization for remotecall. We have to wait for the Future it returns
+# before putting the worker back in the pool.
+function remotecall_pool(rc_f::typeof(remotecall), f, pool::AbstractWorkerPool, args...; kwargs...)
+    worker = take!(pool)
+    local x
+    try
+        x = rc_f(f, worker, args...; kwargs...)
+    catch
+        put!(pool, worker)
+        rethrow()
+    end
+
+    t = Threads.@spawn try
```

Comment:
I'd rebase this to get multithreading testing, and I think it makes sense to keep it in the same threadpool
```suggestion
    t = Threads.@spawn Threads.threadpool() try
```

Reply:
Rebased, and fixed in 89b4cd3.